package/ffi/main.go

```diff
@@ -394,25 +403,51 @@ func javaBuild(ctx context.Context, client *dagger.Client, hostDirectory *dagger
 	// we can do this on the host and then copy the files into the container
 
 	rename := map[string]string{
-		"linux_x86_64":  "linux-x86-64",
-		"linux_arm64":   "linux-aarch64",
-		"darwin_x86_64": "darwin-x86-64",
-		"darwin_arm64":  "darwin-aarch64",
+		"linux_x86_64":      "linux-x86-64",
+		"linux_arm64":       "linux-aarch64",
+		"darwin_x86_64":     "darwin-x86-64",
+		"darwin_arm64":      "darwin-aarch64",
+		"linux_arm64_musl":  "linux-aarch64",
+		"linux_x86_64_musl": "linux-x86-64",
 	}
 
 	for old, new := range rename {
+		// if the directory does not exist, skip it
+		if _, err := os.Stat(fmt.Sprintf("tmp/%s", old)); os.IsNotExist(err) {
+			continue
+		}
+
+		// remove directory if it exists
+		if err := os.RemoveAll(fmt.Sprintf("tmp/%s", new)); err != nil {
+			return err
+		}
+
 		if err := os.Rename(fmt.Sprintf("tmp/%s", old), fmt.Sprintf("tmp/%s", new)); err != nil {
 			return err
 		}
 	}
 
 	container := client.Container().From("gradle:8.5.0-jdk11").
-		WithDirectory("/src", hostDirectory.Directory("flipt-client-java")).
-		WithDirectory("/src/src/main/resources", hostDirectory.Directory("tmp"), dagger.ContainerWithDirectoryOpts{
-			Exclude: defaultExclude,
-		}).
-		WithWorkdir("/src").
-		WithExec([]string{"gradle", "-x", "test", "build"})
+		WithDirectory("/src", hostDirectory.Directory("flipt-client-java"))
+
+	if buildOpts.libc == glibc {
+		container = container.
+			WithDirectory("/src/src/main/resources", hostDirectory.Directory("tmp"), dagger.ContainerWithDirectoryOpts{
+				Exclude: defaultExclude,
+			}).
+			WithWorkdir("/src").
+			WithExec([]string{"gradle", "-x", "test", "build"})
+	} else {
+		container = container.
+			WithDirectory("/src/src/main/resources/linux-aarch64", hostDirectory.Directory("tmp/linux-aarch64"), dagger.ContainerWithDirectoryOpts{
+				Exclude: []string{"**/*.rlib", "**/*.a", "**/*.d"},
+			}).
+			WithDirectory("/src/src/main/resources/linux-x86-64", hostDirectory.Directory("tmp/linux-x86-64"), dagger.ContainerWithDirectoryOpts{
```

Comment:
@markphelps did I miss `tmp/linux_x86_64_musl` here?

Reply:
no i dont think so. above we are renaming `linux_x86_64_musl` which is where the rust engine initially lives to `linux-x86-64` which is where Java expects the file to live:

```
"linux_arm64_musl":  "linux-aarch64",
"linux_x86_64_musl": "linux-x86-64",
```